### PR TITLE
Take 2: Add Jaeger/tracing, Tokio-Console to Sui and wallet runtimes

### PR DESCRIPTION
**Pull request by velvia**
- Uses the new telemetry module from Mysten-infra
- Wallet log output now goes to a separate file; CLI output no longer depends on logs
- Changed some noisy instrumentation to use  `#[instrument]`

## Benchmark results

On main:
```
~/s/m/fastnft (main=) cargo run --release --bin=bench -- --use-move --num-transactions 160000
Total time: 1548046us, items: 160000, tx/sec: 103356.10182126371
```

On this branch, using default logging settings in bench:
```
cargo run --release --bin=bench -- --use-move --num-transactions 160000
Total time: 1566819us, items: 160000, tx/sec: 102117.73025473906
```

### CODE REVIEW

**Comment by velvia** [comment]
@gdanezis @oxade @laura-makdah Sorry need you to approve the PR again, I had to re-create it because of issues with my old fork not working anymore.   Changes since last time: a bunch of setup code was moved to Mysten-infra, and some inline instrumentation moved to `#[instrument]` headers for compactness.  Thanks!

File: deny.toml [hunk 1]
```diff
@@ -55,6 +55,10 @@ ignore = [
     # no safe upgrade is available, but once one is, we should upgrade Move
     # and remove this
     "RUSTSEC-2022-0002",
+    # telemetry component in mysten-infra clashes with crates.io telemetry package
```
Comment (huitseeker) [inline deny.toml:58]: 👍

**Review by huitseeker — APPROVED** [review]
Thanks a lot for pushing this one!
Historic PR at https://github.com/MystenLabs/sui/pull/859 for ref.

File: doc/src/contribute/observability.md [hunk 2]
```diff
@@ -153,8 +171,23 @@ Also notice `elapsed_milliseconds` which logs the duration of each span.
 {"v":0,"name":"sui","msg":"[PROCESS_CERT - END]","level":20,"hostname":"Evan-MLbook.lan","pid":51425,"time":"2022-03-08T22:48:11.248688Z","target":"sui_core::authority_server","line":67,"file":"sui_core/src/authority_server.rs","tx_digest":"t#d1385064287c2ad67e4019dd118d487a39ca91a40e0fd8e678dbc32e112a1493","elapsed_milliseconds":2}
 ```
 
+### Jaeger (seeing distributed traces)
+
+To see nested spans visualized with [Jaeger](https://www.jaegertracing.io), do the following:
+
+1. Run this to get a local Jaeger container: `docker run -d -p6831:6831/udp -p6832:6832/udp -p16686:16686 jaegertracing/all-in-one:latest`
+2. Run sui like this (trace enables the most detailed spans): `SUI_TRACING_ENABLE=1 RUST_LOG="info,sui_core=trace" ./sui start`
```
Comment (Clay-Mysten) [inline doc/src/contribute/observability.md:179]: You should be able to use 1. for each numbered step and let the numbers be dynamically generated. Try it in preview.

Comment (Clay-Mysten) [inline doc/src/contribute/observability.md:179]: Run Sui (capitalize throughout except for commands, paths, etc.)

File: doc/src/contribute/observability.md [hunk 3]
```diff
@@ -153,8 +171,23 @@ Also notice `elapsed_milliseconds` which logs the duration of each span.
...
+3. Run some transfers with wallet, or run the benchmarking tool.
+4. Browse to `http://localhost:16686/` and select "Sui" as the Service
```
Comment (Clay-Mysten) [inline doc/src/contribute/observability.md:181]: and select **Sui** as the *Service*.

File: doc/src/contribute/observability.md [hunk 4]
```diff
@@ -153,8 +171,23 @@ Also notice `elapsed_milliseconds` which logs the duration of each span.
...
+
+NOTE: separate spans (which are not nested) are not connected as a single trace for now.
+
 ### Live async inspection / Tokio Console
 
 [Tokio-console](https://github.com/tokio-rs/console) is an awesome CLI tool designed to analyze and help debug Rust apps using Tokio, in real time!  It relies on a special subscriber.
 
+1. Build Sui using a special flag: `RUSTFLAGS="--cfg tokio_unstable" cargo build`
+2. Start Sui with `SUI_TOKIO_CONSOLE` set to 1
+3. Clone the console repo and `cargo run` to alunch the console
```
Comment (Clay-Mysten) [inline doc/src/contribute/observability.md:191]: to launch (spelled as to alunch)

Comment (Clay-Mysten) [reply doc/src/contribute/observability.md:191]: Also end sentence with a period to match the others.

**Review by Clay-Mysten — APPROVED** [review]
Thanks, Evan! Great work. I have a bunch of tiny suggestions and a couple of errors caught.

**Review by Clay-Mysten — CHANGES_REQUESTED** [review]
Also make sure you have peers test your commands. I will try to get to that in time but don't want to hold you up.

File: doc/src/contribute/observability.md (hunk 2 above)
Comment (velvia) [reply doc/src/contribute/observability.md:179]: True though in my experience it's also harmless, my editor just does it.  However it does work as you say.

**Pull request merged**